src/Command/ImportExisting.php

```diff
@@ -0,0 +1,309 @@
+		// @codingStandardsIgnoreLine
+		'The content was taken down and the user was clearly warned and discouraged from future violations.' => 'taken-down-user-warned',
+		'The user who uploaded the content has been locked.' => 'user-locked',
+	];
+
+	/**
+	 * {@inheritdoc}
+	 *
+	 * @param Registry $doctrine Doctrine.
+	 * @param Connection $old Old Database Connection.
+	 * @param TakedownUtilInterface $takedownUtil Takedown Utility.
+	 */
+	public function __construct(
+		Registry $doctrine,
+		Connection $old,
+		TakedownUtilInterface $takedownUtil
+	) {
+		$this->doctrine = $doctrine;
+		$this->old = $old;
+		$this->takedownUtil = $takedownUtil;
+
+		parent::__construct();
+	}
+
+	/**
+	 * {@inheritdoc}
+	 */
+	protected function configure() {
+		$this->setName( 'app:import:existing' )
+			->setDescription( 'Import Existing Takedown Database' );
+	}
+
+	/**
+	 * {@inheritdoc}
+	 *
+	 * @param InputInterface $input Input
+	 * @param OutputInterface $output Output
+	 *
+	 * @return void
+	 */
+	protected function execute( InputInterface $input, OutputInterface $output ) {
+		$statement = $this->old->executeQuery( "SELECT * FROM centrallog WHERE test = 'N'" );
+		$statement = $this->old->createQueryBuilder()
+			->select( 'cl.*, dmca.*, cp.*, u.*' )
+			->from( 'centrallog', 'cl' )
+			->leftJoin( 'cl', 'dmcatakedowns', 'dmca', 'cl.id = dmca.log_id' )
+			->leftJoin( 'cl', 'ncmecrelease', 'cp', 'cl.id = cp.log_id' )
+			->leftJoin( 'cl', 'user', 'u', 'cl.user = u.user' )
+			->where( "cl.test = 'N'" )
+			->andWhere( "cl.type IN ('Child Protection', 'DMCA')" )
+			->execute();
+
+		$promises = [];
+
+		$em = $this->doctrine->getEntityManager();
+
+		$result = $statement->fetchAll( \PDO::FETCH_NAMED );
+		foreach ( $result as $item ) {
+			// Get the username.
+			if ( !$item['wiki_user'] ) {
+				$item['wiki_user'] = self::USER_NAMES[$item['user'][0]];
+			}
+			$username = $item['wiki_user'];
+
+			$takedown = new Takedown( [
+				'id' => (int)$item['id'][0],
+				'reporter' => [
+					'username' => $username,
+				],
+				'created' => new \DateTime( $item['timestamp'][0] ),
+			] );
+
+			// Get the Page Ids.
+			$pageIds = [];
+			$siteId = null;
+			if ( $item['files_affected'] ) {
+				$urls = unserialize( $item['files_affected'] );
+				if ( $urls ) {
+					$pageIds = array_map( function ( $url ) {
+						$pieces = explode( '/', parse_url( $url,  PHP_URL_PATH ) );
+						return str_replace( ' ', '_', end( $pieces ) );
+					}, $urls );
+
+					$domain = parse_url( $urls[0],  PHP_URL_HOST );
+					$siteId = self::SITE_IDS[$domain];
+					$takedown->setSiteId( $siteId );
+				}
+			}
+
+			if ( $item['project'] ) {
+				$takedown->setSiteId( self::PROJECT_IDS[$item['project']] );
+			}
+
+			$takedown->setPageIds( $pageIds );
+
+			// Get Involved Users.
+			$involvedNames = [];
+			if ( $item['involved_user'] ) {
+				// @codingStandardsIgnoreLine
+				$involvedNames = @unserialize( $item['involved_user'] );
```

Comment:
I completely agree. I did this because in the old database there is either serialized string, or usernames. There's not really a good way to tell if it's serialized or not without just running the serializer and suppressing errors.
https://stackoverflow.com/questions/1369936/check-to-see-if-a-string-is-serialized

Nothing in the new database is serialized, so we wont have to worry about this in the future. :)